src/encoding_implementations/gaussian_cdf.jl

```diff
@@ -54,32 +75,91 @@ julia> decode(encoding, 3)
  0.4
  0.6
 ```
+
+One can also encode the entire vector as an integer.
+
+```jldoctest
+julia> using ComplexityMeasures, Statistics
+
+julia> x = [0.1, 0.4, 0.7, -2.1, 8.0];
+
+julia> μ, σ = mean(x), std(x); encoding = GaussianCDFEncoding(x; μ, σ, c = 2)
+GaussianCDFEncoding(m=5; c=2, μ=1.42, σ=3.840182287340016)
+
+julia> symbol = encode(encoding, x)
+17
+
+julia> decode(encoding, symbol)
+5-element Vector{SVector{1, Float64}}:
+ [0.0]
+ [0.0]
+ [0.0]
+ [0.0]
+ [0.5000000000000001]
+```
 """
-struct GaussianCDFEncoding{T} <: Encoding
+struct GaussianCDFEncoding{m, T, L <: LinearIndices, C <: CartesianIndices, R} <: Encoding
+    m::Int
     c::Int
     σ::T
     μ::T
-    # We require the input data, because we need σ and μ for encoding single values.
-    function GaussianCDFEncoding(; μ::T, σ::T, c::Int = 3) where T
-        new{T}(c, σ, μ)
+
+    # internal fields: LinearIndices/CartesianIndices for encodings/decodings. binencoder
+    # for discretizing the interval [0, 1]
+    linear_indices::L
+    cartesian_indices::C
+    binencoder::R # RectangularBinEncoding
+
+    # The input `m` restricts what length the input scalar/vector can be.
+    function GaussianCDFEncoding(m::Int = 1; μ::T, σ::T, c::Int = 3) where T
+        m >= 1 || throw(ArgumentError("m must be an integer ≥ 1. Got $m."))
+        ranges = tuple([1:c for i in 1:m]...)
+        cartesian_indices = CartesianIndices(ranges)
+        linear_indices = LinearIndices(ranges)
+        L = typeof(linear_indices)
+        C = typeof(cartesian_indices)
+        binencoder = RectangularBinEncoding(FixedRectangularBinning(0, 1, c + 1))
+        R = typeof(binencoder)
+        new{m, T, L, C, R}(m, c, σ, μ, linear_indices, cartesian_indices, binencoder)
     end
 end
+GaussianCDFEncoding(x::AbstractVector; kwargs...) = GaussianCDFEncoding(length(x); kwargs...)
+
+function Base.show(io::IO, e::GaussianCDFEncoding{m, T, L, C}) where {m, T, L, C}
+    c, μ, σ = e.c, e.μ, e.σ
+    print(io, "GaussianCDFEncoding(m=$m; c=$c, μ=$μ, σ=$σ)")
+end
```

Comment:
We should have a generic `show` method for everything instead of adding `show` here. This `show` method is caught by the test coverage. I would argue, let's remove adding any `show` here and add them all in a generic `show` PR. CF #286